tests/helpers/execute.ts

```diff
@@ -31,15 +40,28 @@ interface ExecuteReturn {
   exitCode: number | null;
 }
 
+async function handleProcessExitEvent(stderrPromise: Promise<string>, stdoutPromise: Promise<string>): Promise<void> {
+  await stderrPromise;
+  await stdoutPromise;
+  return Promise.resolve();
+}
+
 export const executeCli = async (args: string[] = [], opts: { env: Env } = { env: undefined }): Promise<ExecuteReturn> => {
   const { env = undefined } = opts;
-  const childProcess = createProcess('./src/index.ts', ['--progress=false', ...args], env);
-  childProcess.stdin.setDefaultEncoding('utf-8');
   const promise = new Promise<ExecuteReturn>((resolve, reject) => {
+    const childProcess = createProcess('./src/index.ts', ['--progress=false', ...args], env);
+    childProcess.stdin.setDefaultEncoding('utf-8');
+    let stdout!: string, stderr!: string;
+    const stdoutPromise = readStream(childProcess.stdout).then((value: undefined | string) => (stdout = value ?? ''));
```

Comment:
according to readStream its return type is string, so why the value in the then clause is "undefined | string"?

Reply:
fixed